engine-rest/engine-rest-openapi/src/main/templates/paths/process-definition/{id}/restart/post.ftl

```diff
@@ -0,0 +1,84 @@
+
+              For more information about the difference between synchronous and asynchronous execution,
+              please refer to the related section of the
+              [User Guide](${docsUrl}/user-guide/process-engine/process-instance-restart/#execution)." />
+
+  "parameters" : [
+
+    <@lib.parameter
+        name = "id"
+        location = "path"
+        type = "string"
+        required = true
+        last = true
+        desc = "The id of the process definition of the process instances to restart."/>
+  ],
+
+  <@lib.requestBody
+      mediaType = "application/json"
+      dto = "RestartProcessInstanceDto"
+      examples = ['"example-1": {
+                     "summary": "Restarting one or more Process Instances with known processInstanceIds",
+                     "value": {
+                       "instructions": [
+                         {
+                           "type": "startAfterActivity",
+                           "activityId": "aUserTask"
+                         }
+                       ],
+                       "processInstanceIds": [
+                         "aProcessInstance",
+                         "anotherProcessInstance"
+                       ],
+                       "initialVariables" : true,
+                       "skipCustomListeners" : true,
+                       "withoutBusinessKey" : true
+                     }
+                   }',
+                   '"example-2": {
+                     "summary": "Restarting one or more Process Instances using a historicProcessInstanceQuery",
+                     "value": {
+                       "instructions": [
+                         {
+                           "type": "startAfterActivity",
+                           "activityId": "aUserTask"
+                         }
+                       ],
+                       "historicProcessInstanceQuery": {
+                         "processDefinitionId": "aProcessDefinitionId",
+                         "processInstanceBusinessKey" : "businessKey"
+                       },
+                       "initialVariables" : true,
+                       "skipCustomListeners" : true,
+                       "withoutBusinessKey" : true
+                     }
+                   }'
+                 ] />
+
+  "responses" : {
+
+    <@lib.response
+        code = "204"
+        desc = "Request successful." />
+
+    <@lib.response
+        code = "400"
+        dto = "ExceptionDto"
+        last = true
+        desc = "In case following parameters are missing: `instructions`, `activityId` or `transitionId`,
+                `processInstanceIds` or `historicProcessInstanceQuery`, an exception of type `InvalidRequestException` is returned. 
+                See the [Introduction](${docsUrl}/reference/rest/overview/#error-handling)
+                for the error response format." />
+
```

Comment:
The following response is missing (it's thrown by the `ProcessDefinitionResourceImpl#getProcessDefinition` methods:
```suggestion
    <@lib.response
        code = "404"
        dto = "ExceptionDto"
        last = true
        desc = "Process definition with given id does not exist. See the
                [Introduction](${docsUrl}/reference/rest/overview/#error-handling)
                for the error response format." />
```

Reply:
it's 500.